Samples/SSAO/include/SSAO.h

```diff
@@ -176,9 +176,9 @@ class _OgreSampleClassExport Sample_SSAO : public SdkSample
             OGRE_EXCEPT(Exception::ERR_NOT_IMPLEMENTED, "Your graphics card does not support vertex and fragment"
                         " programs, so you cannot run this sample. Sorry!", "Sample_SSAO::testCapabilities");
         }
-		if (StringUtil::startsWith(caps->getRenderSystemName(), "OpenGL ES"))
+        if (!StringUtil::startsWith(caps->getRenderSystemName(), "OpenGL 3+") && !StringUtil::startsWith (caps->getRenderSystemName (), "Direct3D11"))
```

Comment:
i downloaded the pre-built samples for 1.12.  it looks like the GL works fine but the D3D9 sample crashes the sample browser.  (so does DX11 without hte SSAOPost.hlsl change above).

on my 1.11.5 build, GL doesn't crash but it shows all black.

this used to work for dx9 but whenever the RTTS demo was changed it must have broken the dx9 and dx11.

you could just take the SSAOPost.hlsl change and drop the SSAO.h change but dx9 is still broken.

Reply:
(i actually just care about the dx9 ssao myself so if it's supposed to work i can take a look offline)

Reply:
so to clarify:  take the SSAOPost.hlsl change to fix dx11.  i'll keep looking into the dx9 failure since it's what i care about anyway :).  finally, drop the SSAO change since i now understand this should work on dx9.

Reply:
update: the 1.12 prebuilt samples don't appear to include cg binaries and plugins so the ssao sample there is dying because there is no suitable fallback material.  not sure why the try/catch isn't handling whatever exception comes along. on my 1.11.5 build it's failing to assemble the cg source code in the dx9 gpu program.  i'm not sure if this is correct or if it should be using a cgprogram class instead.